pkgs/tools/graphics/diagrams-builder/default.nix

```diff
@@ -0,0 +1,43 @@
+/*
+  If user need access to more haskell package for building his
+  diagrams, he simply has to pass these package through the
+  extra packages function as follow in `config.nix`:
+  
+  ~~~
+  diagrams-builder.override {
+    extraPackages = self : [myHaskellPackage];
+  }
+  ­~~~
+*/
+
+{ stdenv, ghcWithPackages, makeWrapper, diagrams-builder, extraPackages ? (self: []) }:
+let
+  # Used same technique as for the yiCustom package.
+  w = ghcWithPackages 
+    (self: [ diagrams-builder ] ++ extraPackages self);
+  wrappedGhc = w.override { ignoreCollisions = true; };
+  ghcVersion = w.version;
+in
+stdenv.mkDerivation {
+  name = "diagrams-builder";
+  buildInputs = [ makeWrapper ];
+  buildCommand = ''
```

Comment:
I'm not sure I can see the potential yet. Do you have an example?

I can definitely make the whole thing more concise by moving 

~~~
makeWrapper \
"${diagrams-builder}/bin/diagrams-builder-svg" "$out/bin/diagrams-builder-svg" \
 --set NIX_GHC ${wrappedGhc}/bin/ghc \
 --set NIX_GHC_LIBDIR ${wrappedGhc}/lib/ghc-${ghcVersion}
~~~

into a function parametrized with the executable suffix that differs.